### ISSUE TITLE
Errors and empty responses must be handled correctly on FE

### ISSUE THREAD

**Issue by smacker** [src-d/code-annotation#104]
right now it's not always true.

- [x] Check tests, make sure it covers `error` path. Add expected result. Fix code if test doesn't pass. #205
- [ ] Check tests, make sure it covers `empty` path. Add expected result. Fix code if test doesn't pass.
- [ ] Mock responses 1 by 1 and make sure UX is good for each such response. Fix code if UX is not good.

**Comment by dpordomingo** [comment]
I saw some of them... but it would be great to have a list...

**Comment by smacker** [comment]
yes. It's part of this issue. Find those places. I would go like this:

1. Check tests, make sure it covers `error` path. Add expected result. Fix code if test doesn't pass.
2. Check tests, make sure it covers `empty` path. Add expected result. Fix code if test doesn't pass.
3. Mock responses 1 by 1 and make sure UX is good for each such response. Fix code if UX is not good.

**Comment by ricardobaeta** [comment]
I'm here to help.

**Comment by dpordomingo** [comment]
@smacker I moved this to `in progress`  because I saw there is a PR addressing some improvement required by this issue.

next steps:
- if there is no pending work to do about this issue &rarr; you can close it :tada: 
- if there is pending work to do:
  - and you're already on it &rarr; keep it as `in progress`,
  - but you're not on it &rarr; move it to `TODO` 

Many thanks!!
